src/NewProject/source_grid.cpp

```diff
@@ -378,3 +439,30 @@ QStringList sourceGrid::GetAllDesignSourceExtentions() const {
   }
   return filters.values();
 }
+
+void sourceGrid::initLanguageCombo(int row, const QVariant &data) {
+  auto combo = CreateLanguageCombo();
+  combo->setCurrentIndex(combo->findData(data));
+  m_tableViewSrc->setIndexWidget(m_model->index(row, LANG_COL_NUM), combo);
+  connect(combo, SIGNAL(currentIndexChanged(int)), this,
+          SLOT(languageHasChanged()));
+}
+
+bool operator==(const FOEDAG::filedata &a, const FOEDAG::filedata &b) {
```

Comment:
Isn't it a default implementation? field by field comparison. why do we need its explicit implementation?